source/test/TestApp/AutoGenTests/AutoGenDataTestData.cpp

```diff
@@ -3,144 +3,270 @@
+        testData.m_InitiateFileUploadsResponse.FromJson(internalResult.ToJson());
 
-    UNREFERENCED_PARAMETER(result);
-    return S_OK;
-}
+        return S_OK;
+    }
 
-void AutoGenDataTests::FillPlayFabDataGetObjectsRequest( PlayFab::DataModels::GetObjectsRequest* request )
-{
-    // TODO: debug  test
-    PlayFab::JsonDocument inputJson;
-    inputJson.Parse( "{ \"Entity\": { \"Id\": \"ABC90712ABC\", \"Type\": \"title_player_account\", \"TypeString\": \"title_player_account\" }}" );
-    request->FromJson(inputJson);
-}
+    void AutoGenDataTests::FillDeleteFilesPrerequisiteFinalizeFileUploadsRequest(PlayFab::DataModels::FinalizeFileUploadsRequest* request)
+    {
+        PlayFab::JsonDocument inputJson;
+        inputJson.Parse("{ \"FileNames\": [ \"TestFileDelete\" ], \"Entity\": { \"Id\": \"B64BE91E5DBD5597\", \"Type\": \"title_player_account\", \"TypeString\": \"title_player_account\" }}");
+        request->FromJson(inputJson);
+    }
 
-HRESULT AutoGenDataTests::ValidatePlayFabDataGetObjectsResponse( PlayFabDataGetObjectsResponse* result )
-{
-    // result.entity = PlayFabEntityKey const*;
-    // result.objects = struct PlayFabDataObjectResultDictionaryEntry const*;
-    // result.objectsCount = uint32_t;
-    // result.profileVersion = int32_t;
+    HRESULT AutoGenDataTests::StoreDeleteFilesPrerequisitePFDataFinalizeFileUploadsResponse(PFDataFinalizeFileUploadsResponse* result)
+    {
+        PlayFab::DataModels::FinalizeFileUploadsResponse internalResult(*result);
+        testData.m_FinalizeFileUploadsResponse.FromJson(internalResult.ToJson());
 
-    UNREFERENCED_PARAMETER(result);
-    return S_OK;
-}
+        return S_OK;
+    }
 
-void AutoGenDataTests::FillPlayFabDataInitiateFileUploadsRequest( PlayFab::DataModels::InitiateFileUploadsRequest* request )
-{
-    // TODO: debug  test
-    PlayFab::JsonDocument inputJson;
-    inputJson.Parse( "{ \"FileNames\": [ \"profile_configuration\", \"game_state\" ], \"Entity\": { \"Id\": \"ID\", \"Type\": \"title_player_account\", \"TypeString\": \"title_player_account\" }}" );
-    request->FromJson(inputJson);
-}
+    void AutoGenDataTests::FillDeleteFilesRequest(PlayFab::DataModels::DeleteFilesRequest* request)
+    {
+        // TODO: debug Failing test
+        PlayFab::JsonDocument inputJson;
+        inputJson.Parse("{ \"FileNames\": [ \"TestFileDelete\" ], \"Entity\": { \"Id\": \"B64BE91E5DBD5597\", \"Type\": \"title_player_account\", \"TypeString\": \"title_player_account\" }}");
+        request->FromJson(inputJson);
+    }
 
-HRESULT AutoGenDataTests::ValidatePlayFabDataInitiateFileUploadsResponse( PlayFabDataInitiateFileUploadsResponse* result )
-{
-    // result.entity = PlayFabEntityKey const*;
-    // result.profileVersion = int32_t;
-    // result.uploadDetails = PlayFabDataInitiateFileUploadMetadata const* const*;
-    // result.uploadDetailsCount = uint32_t;
+    HRESULT AutoGenDataTests::ValidatePFDataDeleteFilesResponse(PFDataDeleteFilesResponse* result)
```

Comment:
I'm curious what this method is supposed to do. Based on the name, I expected some form of validation.

Reply:
These are all manual validation which if I understood John or Jason correctly is needed once. There are two "Pass" states, that I believe are only distinguished when working on the tests...or something? If a call is functioning and sending up proper data but the response from the service has no data, there's PassingButNoData, and then once the input data and/or prerequisites are set up correctly and the response data has real stuff, there's PassingWithData. But both indicate the actual call is functioning correctly. Someone else chime in if I'm missing something, but as I understand it, the response validation seems to be needed only when working in a test.

Reply:
validation isn't always needed, but can be used to confirm the result is as expected.  For example, CreateGroup allows you specify a group name in the request & returns the group name in the response.  You could use the Validate method to confirm that those are identical.  Mostly a sanity check but still useful in some cases.

Reply:
Okay. That was my last question. I think I'm good on this now.